File: code-breaker/typescript/tests/secretBuilder.ts
import { createSecret, Peg, type Secret } from '../src/codeBreaker.js';

export class SecretBuilder {
  private _pegs: readonly [Peg, Peg, Peg, Peg] = [Peg.One, Peg.Two, Peg.Three, Peg.Four];

  with(a: Peg, b: Peg, c: Peg, d: Peg): this {
    this._pegs = [a, b, c, d];
    return this;
  }

  build(): Secret {
    return createSecret(this._pegs);
  }
}

File: code-breaker/python/tests/secret_builder.py
from code_breaker import Peg, Secret


class SecretBuilder:
    def __init__(self) -> None:
        self._pegs = (Peg.ONE, Peg.TWO, Peg.THREE, Peg.FOUR)

    def with_pegs(self, a: Peg, b: Peg, c: Peg, d: Peg) -> "SecretBuilder":
        self._pegs = (a, b, c, d)
        return self

    def build(self) -> Secret:
        return Secret(pegs=self._pegs)
